### PR TITLE
Input validation VR505-1

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

Input Validation of VR505-1

- Resolution must be P1D OR PT1H OR PT15M when ChargeType is Tariff

Unknown value added to Resolution to avoid default values ending up as valid input

## References

* #195

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ResolutionTariffValidationRuleTests.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        [InlineAutoMoqData(Resolution.PT15M, true)]
+        public void ResolutionTariffValidationRule_WithTariffType_Test(
+            Resolution resolution,
+            bool expected,
+            [NotNull] ChargeCommand command)
+        {
+            command.ChargeOperation.Type = ChargeType.Tariff;
+            command.ChargeOperation.Resolution = resolution;
+            var sut = new ResolutionTariffValidationRule(command);
+            sut.IsValid.Should().Be(expected);
```
Comment (bemwen) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ResolutionTariffValidationRuleTests.cs:41]: Not that important in these cases, but according to our normal guidelines you should have Arrange, Act, Assert comments as this is more than 3 lines.

**Review by bemwen — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ResolutionTariffValidationRuleTests.cs (hunk 1 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ResolutionTariffValidationRuleTests.cs]: Added AAA to unittests due to over 3 lines

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/ValidationRuleIdentifier.cs [hunk 2]
```diff
@@ -23,5 +23,6 @@ public enum ValidationRuleIdentifier
         SenderIsMandatory, // VR150
         RecipientIsMandatory, // VR153
         ChargeOperationIdIsRequired, // VR223
+        ResolutionTariff, // VR505-1
```
Comment (lasrinnil) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/ValidationRuleIdentifier.cs:26]: Maybe something like: TariffResolutionIncorrect ?

**Review by lasrinnil — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/ValidationRuleIdentifier.cs (hunk 2 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/ValidationRuleIdentifier.cs]: Changed to InvalidResolutionTypeOnTariffOnCreate

**Pull request merged**